src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -265,23 +265,43 @@ void maybeAddPartitions(int minPartitionNum) {
         if (partitionNum < minPartitionNum) {
           LOG.info("MultiClusterTopicManagementService will increase partition of the topic {} "
               + "in cluster {} from {} to {}.", _topic, _zkConnect, partitionNum, minPartitionNum);
-
+          Collection<Integer> blackListedBrokers =
+              _topicFactory.getBlackListedBrokers(_zkConnect);
           scala.Option<scala.collection.Map<java.lang.Object, scala.collection.Seq<java.lang.Object>>> replicaAssignment = scala.Option.apply(null);
           scala.Option<Seq<Object>> brokerList = scala.Option.apply(null);
-          adminZkClient.addPartitions(_topic, existingAssignment, adminZkClient.getBrokerMetadatas(RackAwareMode.Disabled$.MODULE$, brokerList), minPartitionNum, replicaAssignment, false);
+          Collection<BrokerMetadata> brokers =
+              new HashSet<>(scala.collection.JavaConversions.asJavaCollection(adminZkClient.getBrokerMetadatas(RackAwareMode.Disabled$.MODULE$, brokerList)));
+
+          if (!blackListedBrokers.isEmpty()) {
+            brokers.removeIf(broker -> blackListedBrokers.contains(broker.id()));
+          }
+          adminZkClient.addPartitions(_topic, existingAssignment,
+              scala.collection.JavaConversions.collectionAsScalaIterable(brokers).toSeq(),
+              minPartitionNum, replicaAssignment, false);
         }
       } finally {
         zkClient.close();
       }
     }
 
+    private Collection<Broker> getAvailableBrokers(KafkaZkClient zkClient) {
```

Comment:
Set<Broker> don't want to list a broker more than once.

Reply:
fixed